dallinger/recruiters.py

```diff
@@ -406,6 +409,48 @@ def _create_mturk_qualifications(self):
                 pass
 
 
+# TODO: expiring HITs on shutdown becomes more complicated here
+# because we have to go back and find all the HITs that have been
+# created. One solution could be to expire the last HIT when creating
+# a new one, but then we would have to require n=1 recruitments, which
+# places a constraint on what experimental designs can be used.  E.g.,
+# might want multiple participants will different HITs going in
+# parallel
+class MTurkRobustRecruiter(MTurkRecruiter):
```

Comment:
This doesn't seem particularly high priority given that expiring HITs is currently disabled even in the MTurkRecruiter. When it comes to it, I think it would make more sense to keep a list of all the hits that have been created so that they can all be expired when the experiment ends.

Reply:
I don't see any reason not to replace MTurkRecruiter with this class and to remove the MTurkLargeRecruiter completely. But this might create backwards compatibility issues?

Reply:
It's clearly better than failing silently for the case when the initial recruitment is less than 9 and a subsequent recruitment takes it beyond 9. So I think the key question is whether it creates a backwards compatibility issue for experiments recruiting <9. It obviously changes the interaction with MTurk and what I'm not sure of is whether that there are any side effects I'm not aware of from creating a new HIT rather than extending an existing one. Any thoughts on that, @suchow? If we aren't sure, adding a new recruiter class is certainly the safer option, but I think we should be sure to document the recruiter options so it's clearer how to decide which one to use. And probably we should make the MTurkRecruiter raise an error in a more obvious way if you try to recruit more than MTurk will allow.

Reply:
The backwards compatibility issues I can imagine may be too convoluted to really be worried about, but the type of thing I imagine is someone having written their own scripts that assume all the recruitment requests are associated with one HIT.  Honestly I think the user base of dallinger right now is not large enough to be super concerned about such a pedantic backwards compatibility issue, and it's probably much less confusing to have the default recruiter behave nicely.

Reply:
@suchow or @vlall I think this is up to you to make a judgment call on.

Reply:
Tagged this as shelved. @suchow suggested that there might already be a fix for this. Will close for now and reopen later if this comes up again.

Reply:
What's the existing fix?

Reply:
This doesn't appear to do anything to prevent the same user accepting multiple HITs. Given we have fingerprinting to avoid people gaming the system deliberately by using multiple aounts, I worry that this could be construed as a breach of AWS's AUP:

> Disrupting, manipulating, or impairing the operation of the MTurk website (e.g., don’t try to "game" search results for HITs, or knowingly publish HITs that Workers will be required to return after accepting them)

Perhaps this could be modified to use a qualification to work around it?